### PR TITLE
Service bubble needs to maintain the location

### PR DESCRIPTION
**Pull request by deniss-mc**
**Jira issue:**
https://glia.atlassian.net/browse/MUIC-663

**Additional info:**
Location holder for bubble

### CODE REVIEW

File: widgetssdk/src/main/java/com/glia/widgets/core/chathead/ChatHeadService.java [hunk 1]
```diff
@@ -93,11 +94,18 @@ private void initChatHeadView(
         int screenWidth = displayMetrics.widthPixels;
 
         params.gravity = Gravity.TOP | Gravity.START;
-        params.x = getDefaultXPosition(screenWidth);
-        params.y = getDefaultYPosition(screenHeight);
+
+        Pair<Integer, Integer> savedPosition = controller.onChatHeadPositionRequested();
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/core/chathead/ChatHeadService.java:98]: here `saved` doesn't add useful info, better just `chatHeadPosition`. Whether it's saved or not, how saved etc is controller business.

File: widgetssdk/src/main/java/com/glia/widgets/core/chathead/ChatHeadService.java [hunk 2]
```diff
@@ -75,12 +75,13 @@ private void initChatHeadView(
...
                             layoutParams.y = Float.valueOf(y).intValue();
                             windowManager.updateViewLayout(chatHeadView, layoutParams);
+                            controller.onChatHeadPositionChanged(layoutParams.x, layoutParams.y);
                         },
                         (v) -> controller.onChatHeadClicked()
                 ));
     }
 
-    private WindowManager.LayoutParams getLayoutParams() {
+    private WindowManager.LayoutParams getLayoutParams(ServiceChatBubbleController controller) {
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/core/chathead/ChatHeadService.java:84]: `getLayoutParams` only cares about position so no point to inject the whole controller. improves readability.

File: widgetssdk/src/main/java/com/glia/widgets/view/head/ChatHeadPositionHolder.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package com.glia.widgets.view.head;
+
+import androidx.core.util.Pair;
+
+public class ChatHeadPositionHolder {
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/view/head/ChatHeadPositionHolder.java:5]: why not just pojo like `class ChatHeadPosition` with setX/getX methods? Currently `ChatHeadPositionHolder` makes me think theres something more advanced going on than just plain old java object.

File: widgetssdk/src/main/java/com/glia/widgets/view/head/ChatHeadPositionHolder.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.glia.widgets.view.head;
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/view/head/ChatHeadPositionHolder.java:1]: are ChatHead and ChatBubble the same things? If so then stick to one naming.

File: widgetssdk/src/main/java/com/glia/widgets/view/head/controller/ServiceChatBubbleController.java [hunk 5]
```diff
@@ -107,6 +113,16 @@ public void onApplicationStop() {
         toggleChatBubbleServiceUseCase.execute(null);
     }
 
+    @Override
+    public void onChatHeadPositionChanged(int x, int y) {
+        chatHeadPositionHolder.setPosition(x, y);
+    }
+
+    @Override
+    public Pair<Integer, Integer> onChatHeadPositionRequested() {
```
Comment (jakob-mc) [inline widgetssdk/src/main/java/com/glia/widgets/view/head/controller/ServiceChatBubbleController.java:122]: `onChatHeadPositionRequested` assumes it's more complex, like async call. but irl it's a getter `getChatHeadPoition`.

**Review by jakob-mc — CHANGES_REQUESTED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/view/head/ChatHeadPositionHolder.java (hunk 4 above)
Comment (deniss-mc) [reply widgetssdk/src/main/java/com/glia/widgets/view/head/ChatHeadPositionHolder.java:1]: chat head should be used

**Review by jakob-mc — APPROVED** [review]

**Review by AndriiHorishniiMOC — APPROVED** [review]

**Pull request merged**